### PR TITLE
Maintain page and section id in the CoreStore - consolidate to PagePathways

### PR DESCRIPTION
**Pull request by jovergaag**
## Description of the change

Maintains the current `page` and `section` id in the `CoreStore`. The `page` id is used for both the Pathways and Core layouts, but the `section` id is only used in Pathways. This page/section is is used to populate the `current` metric and visualization on the new consolidated `PagePathways`. 

Adding the filters state to the path will follow in another PR.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Part of #1310

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally
- [ ] E2E have been run for Lantern changes ([Steps in the Readme](https://github.com/Recidiviz/pulse-dashboard#running-e2e-tests))

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: src/core/CoreStore/MetricsStore.ts [hunk 1]
```diff
@@ -46,6 +47,21 @@ export default class MetricsStore {
...
+    const map = {
+      [PATHWAYS_PAGES.prison]: {
+        [PATHWAYS_SECTIONS.populationOverTime]: this.prisonPopulationOverTime,
+      },
+      [PATHWAYS_PAGES.supervision]: {
+        [PATHWAYS_SECTIONS.populationOverTime]: this
+          .supervisionPopulationOverTime,
+      },
+    };
+    // @ts-ignore
```
Comment (jovergaag) [inline src/core/CoreStore/MetricsStore.ts:61]: Once we add the other two transition pages here, se can remove this `ts-ignore`

File: src/withRouteSync.tsx [hunk 2]
```diff
@@ -18,28 +18,45 @@
...
+  } = rawParams;
   const entityId =
     !rawEntityId || rawEntityId === "STATE_DOC"
       ? rawEntityId
       : decrypt(rawEntityId);
+  const pageId = rawPageId as PathwaysPage;
+  const sectionId = rawSectionId as PathwaysSection;
   return {
     entityId: entityId || DEFAULT_ENTITY_ID,
+    pageId,
```
Comment (jovergaag) [inline src/withRouteSync.tsx:58]: Page should never be undefined - either on Pathways or Core layout.

File: src/core/CoreNavigation.tsx [hunk 3]
```diff
@@ -67,18 +64,12 @@ const CoreNavigation: React.FC = () => {
...
       </div>
       <div className="CoreNavigation__right">
-        {flags.enableCoreTabNavigation && (
-          <PageNavigation
-            currentView={currentView}
-            currentPage={currentPage ?? ""}
-            pageOptions={pageOptions}
-          />
-        )}
+        {flags.enableCoreTabNavigation && <PageNavigation />}
```
Comment (Catacola) [inline src/core/CoreNavigation.tsx:72]: `enableCoreTabNavigation` is set to true for everyone and I don't see us changing it back. Would you mind removing it?

File: src/core/CoreStore/index.ts [hunk 4]
```diff
@@ -75,14 +80,20 @@ export default class CoreStore {
...
     this.setView = this.setView.bind(this);
 
+    this.setPage = this.setPage.bind(this);
+
     this.setSection = this.setSection.bind(this);
   }
 
-  setView(pathname: string): void {
-    this.view = getViewFromPathname(pathname);
+  setView(view: CoreView | PathwaysView): void {
```
Comment (Catacola) [inline src/core/CoreStore/index.ts:88]: Super minor readability nit. All of the other places where we combine pathways and core types in this file have pathways first.
```suggestion
  setView(view: PathwaysView | CoreView): void {
```

File: src/core/MetricVizMapper/MetricVizMapper.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+import VizPopulationOverTime from "../VizPopulationOverTime";
+
+type MetricVizMapperProps = {
+  metric: PathwaysMetric<MetricRecord>;
+};
+
+const MetricVizMapper: React.FC<MetricVizMapperProps> = ({ metric }) => {
+  if (metric instanceof PopulationOverTimeMetric) {
+    return <VizPopulationOverTime metric={metric} />;
+  }
```
Comment (Catacola) [inline src/core/MetricVizMapper/MetricVizMapper.tsx:32]: This abstraction is going to be so nice!

File: src/core/PageNavigation.tsx [hunk 6]
```diff
@@ -44,21 +45,20 @@ const PageNavigation: React.FC<propTypes> = ({
           currentView && currentView.toLowerCase() !== "pathways",
       })}
     >
-      {pageOptions.map((page) => (
-        <li key={page}>
+      {pageOptions.map((pageOption: any) => (
```
Comment (Catacola) [inline src/core/PageNavigation.tsx:48]:
```suggestion
      {pageOptions.map((pageOption: keyof Navigation) => (
```

Or we could create a new type in `types/navigation.ts`:
```
export type NavigationSection = keyof Navigation;
```

File: src/core/PageNavigation.tsx [hunk 7]
```diff
@@ -44,21 +45,20 @@ const PageNavigation: React.FC<propTypes> = ({
...
+        <li key={pageOption}>
           <Link
-            to={`/${currentView}/${page}`}
+            to={`/${currentView}/${pageOption}`}
             className={cx("PageNavigation__option", {
               "PageNavigation__option--selected":
-                (currentPage && currentPage.toLowerCase()) === page,
+                (page && page.toLowerCase()) === pageOption,
```
Comment (Catacola) [inline src/core/PageNavigation.tsx:54]: Nit:

```suggestion
                (page?.toLowerCase()) === pageOption,
```

File: src/core/PagePathways/PagePathways.scss [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+// =============================================================================
+
+@import "../CoreConstants";
```
Comment (Catacola) [inline src/core/PagePathways/PagePathways.scss:18]: Is this import needed?

File: src/core/PathwaysNavigation/PathwaysNavigation.tsx [hunk 9]
```diff
@@ -19,28 +19,17 @@ import "./PathwaysNavigation.scss";
...
-  const { currentTenantId, userStore } = useRootStore();
+  const { currentTenantId } = useRootStore();
   if (!currentTenantId) return null;
-
-  const navigationLayout = userStore.userAllowedNavigation;
-  const [currentView, currentPage] = pathname.split("/").slice(1, 3);
-  const pageOptions = navigationLayout[currentView] ?? [];
-
   return (
     <>
```
Comment (Catacola) [inline src/core/PathwaysNavigation/PathwaysNavigation.tsx:30]: Do we still need these `<>` wrapping the component? It looks like there's just one top-level tag: `<nav>`.

File: src/core/VizPopulationOverTime/VizPopulationOverTime.tsx [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+        isError={error}
+      />
+      <PopulationTimeSeriesChart
+        metric={metric}
+        title={chartTitle}
+        data={dataSeries}
+      />
+    </>
+  );
+};
```
Comment (Catacola) [inline src/core/VizPopulationOverTime/VizPopulationOverTime.tsx:49]: I just want to comment again about how much I like this abstraction.

File: src/core/hooks/usePageContent.ts [hunk 11]
```diff
@@ -22,24 +22,24 @@ import { useRootStore } from "../../components/StoreProvider";
...
 import { PageContent, StateSpecificPageCopy } from "../content/types";
-import { PathwaysPageId } from "../views";
+import { PathwaysPage } from "../views";
 
 const contentOverrides: { [category: string]: StateSpecificPageCopy } = {
   US_ID: IdContent,
 };
 
-export default function usePageContent(viewName: PathwaysPageId): PageContent {
+export default function usePageContent(pageId: PathwaysPage): PageContent {
```
Comment (Catacola) [inline src/core/hooks/usePageContent.ts:31]: Thanks for this cleanup. I must have missed it in a refactor.

**Review by Catacola — APPROVED** [review]
This all looks fantastic! This abstraction is going to be very useful. Despite being a largish PR, this was quite easy to follow, which bodes well for the future.

I have a bunch of minor comments, but absolutely nothing that is blocking approval. Thanks again!

File: src/core/VizPopulationOverTime/VizPopulationOverTime.tsx (hunk 10 above)
Comment (jovergaag) [reply src/core/VizPopulationOverTime/VizPopulationOverTime.tsx:49]: Heavily inspired by Ian's work in spotlight!

**Pull request merged**